Fails to build with CLANG on GNU/Linux

Latest GIT code and latest stable Shipwright (9.2.3) builds fine with GCC on GNU/Linux, but not with CLANG (clang version 21.1.8).

This is how I configure the code, nothing fancy:
```
mkdir b4
cd b4

cmake --fresh -H.. -B. -GNinja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang
```

Then I build with `ninja`

These are the errors I get:
```
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:32:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/Gui.h:13:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/ConsoleWindow.h:11:
In file included from /usr/include/spdlog/spdlog.h:15:
/usr/include/spdlog/logger.h:334:9: error: call to consteval function 'fmt::basic_format_string<char, const char *, const char *&, int &>::basic_format_string<FMT_COMPILE_STRING, 0>' is not a constant expression
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^
/usr/include/spdlog/logger.h:80:9: note: in instantiation of function template specialization 'spdlog::logger::log_<int &>' requested here
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ^
/home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:491:9: note: in instantiation of function template specialization 'spdlog::logger::log<int &>' requested here
  491 |         SPDLOG_ERROR("ImportTextureRgba16: null texture address for tile {}", tile);
      |         ^
/usr/include/spdlog/spdlog.h:338:31: note: expanded from macro 'SPDLOG_ERROR'
  338 |     #define SPDLOG_ERROR(...) SPDLOG_LOGGER_ERROR(spdlog::default_logger_raw(), __VA_ARGS__)
      |                               ^
/usr/include/spdlog/spdlog.h:337:9: note: expanded from macro 'SPDLOG_LOGGER_ERROR'
  337 |         SPDLOG_LOGGER_CALL(logger, spdlog::level::err, __VA_ARGS__)
      |         ^
/usr/include/spdlog/spdlog.h:293:19: note: expanded from macro 'SPDLOG_LOGGER_CALL'
  293 |         (logger)->log(spdlog::source_loc{__FILE__, __LINE__, SPDLOG_FUNCTION}, level, __VA_ARGS__)
      |                   ^
/usr/include/fmt/core.h:686:54: note: subexpression not valid in a constant expression
  686 |     format_str_.remove_prefix(detail::to_unsigned(it - begin()));
      |                                                   ~~~^~~~~~~~~
/usr/include/fmt/core.h:2637:5: note: in call to 'this->context_.advance_to(&"{} [{}({})]"[1])'
 2637 |     context_.advance_to(begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2632:5: note: in call to 'this->on_format_specs(0, &"{} [{}({})]"[1], &"{} [{}({})]"[1])'
 2632 |     on_format_specs(id, begin, begin);  // Call parse() on empty specs.
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2457:5: note: in call to 'handler.on_replacement_field(0, &"{} [{}({})]"[1])'
 2457 |     handler.on_replacement_field(handler.on_arg_id(), begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2489:21: note: in call to 'parse_replacement_field<char, fmt::detail::format_string_checker<char, const char *, const char *, int> &>(&"{} [{}({})]"[1], &"{} [{}({})]"[11], checker(s))'
 2489 |         begin = p = parse_replacement_field(p - 1, end, handler);
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2740:7: note: in call to 'parse_format_string<true, char, fmt::detail::format_string_checker<char, const char *, const char *, int>>({&"{} [{}({})]"[0], 11}, checker(s))'
 2740 |       detail::parse_format_string<true>(str_, checker(s));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:334:9: note: in call to 'basic_format_string<FMT_COMPILE_STRING, 0>([] {
    struct FMT_COMPILE_STRING : fmt::detail::compile_string {
        using char_type [[maybe_unused]] = fmt::remove_cvref_t<decltype("{} [{}({})]"[0])>;
    };
    return FMT_COMPILE_STRING();
}())'
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1883 |     /* Use the hidden visibility as a workaround for a GCC bug (#1973). */    \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1884 |     /* Use a macro-like name to avoid shadowing warnings. */                  \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1885 |     struct FMT_VISIBILITY("hidden") FMT_COMPILE_STRING : base {               \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1886 |       using char_type FMT_MAYBE_UNUSED = fmt::remove_cvref_t<decltype(s[0])>; \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1887 |       FMT_MAYBE_UNUSED FMT_CONSTEXPR explicit                                 \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1888 |       operator fmt::basic_string_view<char_type>() const {                    \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1889 |         return fmt::detail_exported::compile_string_to_view<char_type>(s);    \
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1890 |       }                                                                       \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1891 |     };                                                                        \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1892 |     return FMT_COMPILE_STRING();                                              \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1893 |   }()
      |   ~~~
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:32:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/Gui.h:13:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/ConsoleWindow.h:11:
In file included from /usr/include/spdlog/spdlog.h:15:
/usr/include/spdlog/logger.h:334:9: error: call to consteval function 'fmt::basic_format_string<char, const char *, const char *&, int &>::basic_format_string<FMT_COMPILE_STRING, 0>' is not a constant expression
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^
/usr/include/spdlog/logger.h:80:9: note: in instantiation of function template specialization 'spdlog::logger::log_<unsigned char &>' requested here
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ^
/home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:998:17: note: in instantiation of function template specialization 'spdlog::logger::log<unsigned char &>' requested here
  998 |                 SPDLOG_ERROR("RGBA Texture that isn't 16 or 32 bit. Size = {}", siz);
      |                 ^
/usr/include/spdlog/spdlog.h:338:31: note: expanded from macro 'SPDLOG_ERROR'
  338 |     #define SPDLOG_ERROR(...) SPDLOG_LOGGER_ERROR(spdlog::default_logger_raw(), __VA_ARGS__)
      |                               ^
/usr/include/spdlog/spdlog.h:337:9: note: expanded from macro 'SPDLOG_LOGGER_ERROR'
  337 |         SPDLOG_LOGGER_CALL(logger, spdlog::level::err, __VA_ARGS__)
      |         ^
/usr/include/spdlog/spdlog.h:293:19: note: expanded from macro 'SPDLOG_LOGGER_CALL'
  293 |         (logger)->log(spdlog::source_loc{__FILE__, __LINE__, SPDLOG_FUNCTION}, level, __VA_ARGS__)
      |                   ^
/usr/include/fmt/core.h:686:54: note: subexpression not valid in a constant expression
  686 |     format_str_.remove_prefix(detail::to_unsigned(it - begin()));
      |                                                   ~~~^~~~~~~~~
/usr/include/fmt/core.h:2637:5: note: in call to 'this->context_.advance_to(&"{} [{}({})]"[1])'
 2637 |     context_.advance_to(begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2632:5: note: in call to 'this->on_format_specs(0, &"{} [{}({})]"[1], &"{} [{}({})]"[1])'
 2632 |     on_format_specs(id, begin, begin);  // Call parse() on empty specs.
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2457:5: note: in call to 'handler.on_replacement_field(0, &"{} [{}({})]"[1])'
 2457 |     handler.on_replacement_field(handler.on_arg_id(), begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2489:21: note: in call to 'parse_replacement_field<char, fmt::detail::format_string_checker<char, const char *, const char *, int> &>(&"{} [{}({})]"[1], &"{} [{}({})]"[11], checker(s))'
 2489 |         begin = p = parse_replacement_field(p - 1, end, handler);
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2740:7: note: in call to 'parse_format_string<true, char, fmt::detail::format_string_checker<char, const char *, const char *, int>>({&"{} [{}({})]"[0], 11}, checker(s))'
 2740 |       detail::parse_format_string<true>(str_, checker(s));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:334:9: note: in call to 'basic_format_string<FMT_COMPILE_STRING, 0>([] {
    struct FMT_COMPILE_STRING : fmt::detail::compile_string {
        using char_type [[maybe_unused]] = fmt::remove_cvref_t<decltype("{} [{}({})]"[0])>;
    };
    return FMT_COMPILE_STRING();
}())'
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1883 |     /* Use the hidden visibility as a workaround for a GCC bug (#1973). */    \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1884 |     /* Use a macro-like name to avoid shadowing warnings. */                  \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1885 |     struct FMT_VISIBILITY("hidden") FMT_COMPILE_STRING : base {               \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1886 |       using char_type FMT_MAYBE_UNUSED = fmt::remove_cvref_t<decltype(s[0])>; \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1887 |       FMT_MAYBE_UNUSED FMT_CONSTEXPR explicit                                 \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1888 |       operator fmt::basic_string_view<char_type>() const {                    \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1889 |         return fmt::detail_exported::compile_string_to_view<char_type>(s);    \
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1890 |       }                                                                       \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1891 |     };                                                                        \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1892 |     return FMT_COMPILE_STRING();                                              \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1893 |   }()
      |   ~~~
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:32:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/Gui.h:13:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/ConsoleWindow.h:11:
In file included from /usr/include/spdlog/spdlog.h:15:
/usr/include/spdlog/logger.h:334:9: error: call to consteval function 'fmt::basic_format_string<char, const char *, const char *&, int &>::basic_format_string<FMT_COMPILE_STRING, 0>' is not a constant expression
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^
/usr/include/spdlog/logger.h:80:9: note: in instantiation of function template specialization 'spdlog::logger::log_<unsigned int &, unsigned int &>' requested here
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ^
/home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:2807:13: note: in instantiation of function template specialization 'spdlog::logger::log<unsigned int &, unsigned int &>' requested here
 2807 |             SPDLOG_WARN("CLOSE_DISPS without matching open {}:{}", p, l);
      |             ^
/usr/include/spdlog/spdlog.h:329:30: note: expanded from macro 'SPDLOG_WARN'
  329 |     #define SPDLOG_WARN(...) SPDLOG_LOGGER_WARN(spdlog::default_logger_raw(), __VA_ARGS__)
      |                              ^
/usr/include/spdlog/spdlog.h:328:9: note: expanded from macro 'SPDLOG_LOGGER_WARN'
  328 |         SPDLOG_LOGGER_CALL(logger, spdlog::level::warn, __VA_ARGS__)
      |         ^
/usr/include/spdlog/spdlog.h:293:19: note: expanded from macro 'SPDLOG_LOGGER_CALL'
  293 |         (logger)->log(spdlog::source_loc{__FILE__, __LINE__, SPDLOG_FUNCTION}, level, __VA_ARGS__)
      |                   ^
/usr/include/fmt/core.h:686:54: note: subexpression not valid in a constant expression
  686 |     format_str_.remove_prefix(detail::to_unsigned(it - begin()));
      |                                                   ~~~^~~~~~~~~
/usr/include/fmt/core.h:2637:5: note: in call to 'this->context_.advance_to(&"{} [{}({})]"[1])'
 2637 |     context_.advance_to(begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2632:5: note: in call to 'this->on_format_specs(0, &"{} [{}({})]"[1], &"{} [{}({})]"[1])'
 2632 |     on_format_specs(id, begin, begin);  // Call parse() on empty specs.
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2457:5: note: in call to 'handler.on_replacement_field(0, &"{} [{}({})]"[1])'
 2457 |     handler.on_replacement_field(handler.on_arg_id(), begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2489:21: note: in call to 'parse_replacement_field<char, fmt::detail::format_string_checker<char, const char *, const char *, int> &>(&"{} [{}({})]"[1], &"{} [{}({})]"[11], checker(s))'
 2489 |         begin = p = parse_replacement_field(p - 1, end, handler);
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2740:7: note: in call to 'parse_format_string<true, char, fmt::detail::format_string_checker<char, const char *, const char *, int>>({&"{} [{}({})]"[0], 11}, checker(s))'
 2740 |       detail::parse_format_string<true>(str_, checker(s));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:334:9: note: in call to 'basic_format_string<FMT_COMPILE_STRING, 0>([] {
    struct FMT_COMPILE_STRING : fmt::detail::compile_string {
        using char_type [[maybe_unused]] = fmt::remove_cvref_t<decltype("{} [{}({})]"[0])>;
    };
    return FMT_COMPILE_STRING();
}())'
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1883 |     /* Use the hidden visibility as a workaround for a GCC bug (#1973). */    \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1884 |     /* Use a macro-like name to avoid shadowing warnings. */                  \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1885 |     struct FMT_VISIBILITY("hidden") FMT_COMPILE_STRING : base {               \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1886 |       using char_type FMT_MAYBE_UNUSED = fmt::remove_cvref_t<decltype(s[0])>; \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1887 |       FMT_MAYBE_UNUSED FMT_CONSTEXPR explicit                                 \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1888 |       operator fmt::basic_string_view<char_type>() const {                    \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1889 |         return fmt::detail_exported::compile_string_to_view<char_type>(s);    \
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1890 |       }                                                                       \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1891 |     };                                                                        \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1892 |     return FMT_COMPILE_STRING();                                              \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1893 |   }()
      |   ~~~
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:32:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/Gui.h:13:
In file included from /home/manuel/src/Shipwright-9.2.3/libultraship/src/../include/ship/window/gui/ConsoleWindow.h:11:
In file included from /usr/include/spdlog/spdlog.h:15:
/usr/include/spdlog/logger.h:334:9: error: call to consteval function 'fmt::basic_format_string<char, const char *, const char *&, int &>::basic_format_string<FMT_COMPILE_STRING, 0>' is not a constant expression
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^
/usr/include/spdlog/logger.h:80:9: note: in instantiation of function template specialization 'spdlog::logger::log_<unsigned char, unsigned int>' requested here
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ^
/home/manuel/src/Shipwright-9.2.3/libultraship/src/fast/interpreter.cpp:4149:13: note: in instantiation of function template specialization 'spdlog::logger::log<unsigned char, unsigned int>' requested here
 4149 |             SPDLOG_CRITICAL("Unhandled OP code: 0x{:X}, for loaded ucode: {}", (uint8_t)opcode,
      |             ^
/usr/include/spdlog/spdlog.h:347:34: note: expanded from macro 'SPDLOG_CRITICAL'
  347 |     #define SPDLOG_CRITICAL(...) SPDLOG_LOGGER_CRITICAL(spdlog::default_logger_raw(), __VA_ARGS__)
      |                                  ^
/usr/include/spdlog/spdlog.h:346:9: note: expanded from macro 'SPDLOG_LOGGER_CRITICAL'
  346 |         SPDLOG_LOGGER_CALL(logger, spdlog::level::critical, __VA_ARGS__)
      |         ^
/usr/include/spdlog/spdlog.h:293:19: note: expanded from macro 'SPDLOG_LOGGER_CALL'
  293 |         (logger)->log(spdlog::source_loc{__FILE__, __LINE__, SPDLOG_FUNCTION}, level, __VA_ARGS__)
      |                   ^
/usr/include/fmt/core.h:686:54: note: subexpression not valid in a constant expression
  686 |     format_str_.remove_prefix(detail::to_unsigned(it - begin()));
      |                                                   ~~~^~~~~~~~~
/usr/include/fmt/core.h:2637:5: note: in call to 'this->context_.advance_to(&"{} [{}({})]"[1])'
 2637 |     context_.advance_to(begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2632:5: note: in call to 'this->on_format_specs(0, &"{} [{}({})]"[1], &"{} [{}({})]"[1])'
 2632 |     on_format_specs(id, begin, begin);  // Call parse() on empty specs.
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2457:5: note: in call to 'handler.on_replacement_field(0, &"{} [{}({})]"[1])'
 2457 |     handler.on_replacement_field(handler.on_arg_id(), begin);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2489:21: note: in call to 'parse_replacement_field<char, fmt::detail::format_string_checker<char, const char *, const char *, int> &>(&"{} [{}({})]"[1], &"{} [{}({})]"[11], checker(s))'
 2489 |         begin = p = parse_replacement_field(p - 1, end, handler);
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/core.h:2740:7: note: in call to 'parse_format_string<true, char, fmt::detail::format_string_checker<char, const char *, const char *, int>>({&"{} [{}({})]"[0], 11}, checker(s))'
 2740 |       detail::parse_format_string<true>(str_, checker(s));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:334:9: note: in call to 'basic_format_string<FMT_COMPILE_STRING, 0>([] {
    struct FMT_COMPILE_STRING : fmt::detail::compile_string {
        using char_type [[maybe_unused]] = fmt::remove_cvref_t<decltype("{} [{}({})]"[0])>;
    };
    return FMT_COMPILE_STRING();
}())'
  334 |         SPDLOG_LOGGER_CATCH(loc)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:34:46: note: expanded from macro 'SPDLOG_LOGGER_CATCH'
   34 |                 err_handler_(fmt_lib::format(SPDLOG_FMT_STRING("{} [{}({})]"), ex.what(), \
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/common.h:55:46: note: expanded from macro 'SPDLOG_FMT_STRING'
   55 |     #define SPDLOG_FMT_STRING(format_string) FMT_STRING(format_string)
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1905:23: note: expanded from macro 'FMT_STRING'
 1905 | #define FMT_STRING(s) FMT_STRING_IMPL(s, fmt::detail::compile_string, )
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/format.h:1882:3: note: expanded from macro 'FMT_STRING_IMPL'
 1882 |   [] {                                                                        \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1883 |     /* Use the hidden visibility as a workaround for a GCC bug (#1973). */    \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1884 |     /* Use a macro-like name to avoid shadowing warnings. */                  \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1885 |     struct FMT_VISIBILITY("hidden") FMT_COMPILE_STRING : base {               \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1886 |       using char_type FMT_MAYBE_UNUSED = fmt::remove_cvref_t<decltype(s[0])>; \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1887 |       FMT_MAYBE_UNUSED FMT_CONSTEXPR explicit                                 \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1888 |       operator fmt::basic_string_view<char_type>() const {                    \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1889 |         return fmt::detail_exported::compile_string_to_view<char_type>(s);    \
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1890 |       }                                                                       \
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1891 |     };                                                                        \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1892 |     return FMT_COMPILE_STRING();                                              \
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1893 |   }()
      |   ~~~
4 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -198,10 +198,26 @@ find_package(ZLIB REQUIRED)
 
 include(FetchContent)
 
+# spdlog runs fmt's consteval format check in every log macro, and clang >= 21 rejects it as
+# not constant with fmt < 11.1: FMT_STRING's compile string converts to two distinct literal
+# objects, so the pointer subtraction fmt does while parsing crosses objects (#7017). Ignore a
+# system spdlog built against such an fmt; the fetched one bundles fmt 12.
+set(_use_system_spdlog ON)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+   AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "21.0")
+    find_package(fmt QUIET)
+    if(fmt_FOUND AND fmt_VERSION VERSION_LESS "11.1")
+        message(STATUS "clang ${CMAKE_CXX_COMPILER_VERSION} can't build against fmt ${fmt_VERSION}, fetching spdlog")
+        set(_use_system_spdlog OFF)
+    endif()
+endif()
+
 # Workaround: torch declares spdlog without OVERRIDE_FIND_PACKAGE and the first declaration
 # wins, which leaves libultraship's find_package(spdlog REQUIRED) with no config to find.
 # Declaring it here first keeps that working. See HarbourMasters/Torch#233.
-find_package(spdlog QUIET)
+if(_use_system_spdlog)
+    find_package(spdlog QUIET)
+endif()
 if(NOT spdlog_FOUND)
     FetchContent_Declare(
         spdlog

